src/figuresClasses.ts

```diff
@@ -1,11 +1,100 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = 'triangle';
+
+  color: string;
+
+  private a: number;
+
+  private b: number;
+
+  private c: number;
+
+  constructor(color: string, a: number, b: number, c: number) {
+    this.color = color;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than zero');
+    }
+
+    const sides = [a, b, c].sort((x, y) => y - x);
+
+    if (sides[0] >= sides[1] + sides[2]) {
+      throw new Error(
+        'The longest side of a triangle cannot be greater' +
+          'than or equal to the sum of the other two sides',
+      );
+    }
+
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
[CODE KNOWLEDGE] - You don't need to manually assign constructor parameters to class properties if you use access modifiers in the constructor parameters. TypeScript will automatically create and assign them. Consider refactoring the constructor parameters to include access modifiers and remove the manual assignments.

GOOD EXAMPLE:
constructor(public color: string, private a: number, private b: number, private c: number) { ... }